arch/RISCV/RISCVInstPrinter.c

```diff
@@ -51,22 +51,65 @@ static const char *getRegisterName(unsigned RegNo, unsigned AltIdx);
 #define PRINT_ALIAS_INSTR
 #include "RISCVGenAsmWriter.inc"
 
+
+static void fixDetailOfEffectiveAddr(MCInst *MI)
+{
+	unsigned reg = 0;
+	int64_t imm = 0;
+
+	assert(3 == MI->flat_insn->detail->riscv.op_count);
+	assert(RISCV_OP_REG == MI->flat_insn->detail->riscv.operands[0].type);
+
+	if (RISCV_OP_IMM == MI->flat_insn->detail->riscv.operands[1].type) {
+		assert(RISCV_OP_REG == MI->flat_insn->detail->riscv.operands[2].type);
+		imm = MI->flat_insn->detail->riscv.operands[1].imm;
+		reg = MI->flat_insn->detail->riscv.operands[2].reg;
+	} else if (RISCV_OP_REG == MI->flat_insn->detail->riscv.operands[1].type) {
+		assert(RISCV_OP_IMM == MI->flat_insn->detail->riscv.operands[2].type);
+		reg = MI->flat_insn->detail->riscv.operands[1].reg;
+		imm = MI->flat_insn->detail->riscv.operands[2].imm;
+	}
+
+	// set up effective address.
+	MI->flat_insn->detail->riscv.operands[1].type = RISCV_OP_MEM;
+	MI->flat_insn->detail->riscv.op_count--;
+     	MI->flat_insn->detail->riscv.operands[1].mem.base = reg;
+     	MI->flat_insn->detail->riscv.operands[1].mem.disp = imm;
+
+	return;
+}
+
+
 //void RISCVInstPrinter::printInst(const MCInst *MI, raw_ostream &O,
 //                                 StringRef Annot, const MCSubtargetInfo &STI) 
 void RISCV_printInst(MCInst *MI, SStream *O, void *info) 
 {
   	MCRegisterInfo *MRI = (MCRegisterInfo *) info;
   	//bool Res = false;
-  	MCInst *NewMI = MI;
+  	//MCInst *NewMI = MI;
   	// TODO: RISCV compressd instructions.
   	//MCInst UncompressedMI;
   	//if (!NoAliases)
     	//Res = uncompressInst(UncompressedMI, *MI, MRI, STI);
   	//if (Res)
     	//NewMI = const_cast<MCInst *>(&UncompressedMI);
-  	if (/*NoAliases ||*/ !printAliasInstr(NewMI, O, info))
-    		printInstruction(NewMI, O, MRI);
+  	if (/*NoAliases ||*/ !printAliasInstr(MI, O, info))
+    		printInstruction(MI, O, MRI);
   		//printAnnotation(O, Annot);
+	// fix load/store type insttuction
+	if (MI->flat_insn->detail->riscv.need_effective_addr)
```

Comment:
the crash happens here: you dereference a NULL pointer `MI->flat_insn->detail`, without checking if detail mode is ON or not.